app/src/main/java/ryey/easer/skills/usource/do_not_disturb/DoNotDisturbUSourceData.java

```diff
@@ -0,0 +1,142 @@
+import androidx.annotation.RequiresApi;
+
+import com.orhanobut.logger.Logger;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Arrays;
+
+import ryey.easer.commons.local_skill.IllegalStorageDataException;
+import ryey.easer.commons.local_skill.dynamics.Dynamics;
+import ryey.easer.commons.local_skill.usource.USourceData;
+import ryey.easer.plugin.PluginDataFormat;
+
+@RequiresApi(api = Build.VERSION_CODES.M)
+public class DoNotDisturbUSourceData implements USourceData {
+
+    static int[] doNotDisturbIDs = {
+            NotificationManager.INTERRUPTION_FILTER_UNKNOWN,
+            NotificationManager.INTERRUPTION_FILTER_ALL,
+            NotificationManager.INTERRUPTION_FILTER_PRIORITY,
+            NotificationManager.INTERRUPTION_FILTER_NONE,
+            NotificationManager.INTERRUPTION_FILTER_ALARMS,
+    };
+    boolean[] doNotDisturbModes = new boolean[doNotDisturbIDs.length];
+
+    private static final String T_doNotDisturb = "doNotDisturb";
+
+    DoNotDisturbUSourceData(@NonNull String data, @NonNull PluginDataFormat format, int version) throws IllegalStorageDataException {
+        try {
+            JSONObject jsonObject = new JSONObject(data);
+            JSONArray modes = jsonObject.optJSONArray(T_doNotDisturb);
+            if (modes != null) {
+                for (int i = 0; i < doNotDisturbIDs.length; i++) {
+                    this.doNotDisturbModes[doNotDisturbIDs[i]] = (modes.getInt(i + doNotDisturbIDs.length) == 1);
+                }
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+            throw new IllegalStorageDataException(e);
+        }
+    }
+
+    DoNotDisturbUSourceData(boolean[] doNotDisturbModes) {
+        this.doNotDisturbModes = doNotDisturbModes;
+    }
+
+    @NonNull
+    @Override
+    public String serialize(@NonNull PluginDataFormat format) {
+        JSONObject jsonObject = new JSONObject();
+        try {
+            JSONArray modes = new JSONArray();
+            for (int i = 0; i < doNotDisturbIDs.length; i++) {
+                modes.put(i, doNotDisturbIDs[i]);
+                modes.put(i + doNotDisturbIDs.length, (this.doNotDisturbModes[doNotDisturbIDs[i]] ? 1 : 0));
+            }
+            jsonObject.put(T_doNotDisturb, modes);
+        } catch (JSONException e) {
+            Logger.e(e, "Error putting %s data", getClass().getSimpleName());
+            e.printStackTrace();
+        }
+        return jsonObject.toString();
+    }
+
+    @Override
+    public boolean isValid() {
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this)
+            return true;
+        if (!(obj instanceof DoNotDisturbUSourceData))
+            return false;
+        return Arrays.equals(((DoNotDisturbUSourceData) obj).doNotDisturbModes, doNotDisturbModes);
+    }
+
+    public boolean match(int doNotDisturbMode) {
```

Comment:
Similar to #451, probably not to have this method? But I see the point of having this method in this particular class, because of the way how modes are stored.